### PR TITLE
JBPM-5714 - Handle adding new app launcher once Case Management Showc…

### PR DESCRIPTION
**Pull request by cristianonicolai**
…ase is provisioned

Add provisioning lifecycle events
Add provisioning status
Rename package cases to casemgmt to match jBPM core API

### CODE REVIEW

**Comment by cristianonicolai** [comment]
Depends on https://github.com/droolsjbpm/kie-wb-common/pull/703

**Comment by cristianonicolai** [comment]
@manstis mind checking this one as well.

File: jbpm-wb-integration/jbpm-wb-integration-backend/src/main/java/org/jbpm/workbench/wi/backend/server/casemgmt/service/CaseProvisioningExecutor.java [hunk 1]
```diff
@@ -61,6 +79,12 @@ protected int getManagementInterfaceStatus(String host, String managementPort, S
         }
     }
 
+    private interface ProvisioningCommand {
```
Comment (manstis) [inline jbpm-wb-integration/jbpm-wb-integration-backend/src/main/java/org/jbpm/workbench/wi/backend/server/casemgmt/service/CaseProvisioningExecutor.java]: Do we really need another ```Command``` interface? There are quite a few around already.

**Comment by cristianonicolai** [comment]
@mantis I duplicated the command interface to avoid pulling uberfire
dependencies into the integration test.

**Pull request merged**